### PR TITLE
Adding input and output data information to models

### PR DESCRIPTION
**Pull request by elisehuard**
First stab at getting started on the data items (input and output)

### CODE REVIEW

File: src/witan/app/test_data.clj [hunk 1]
```diff
@@ -10,13 +11,37 @@
...
+                              :input-data [{:category "long population"}
+                                           {:category "overlay housing"}
+                                           {:category "trend data"}]
+                              :output-data [{:category "housing-linked population"}]})
+        ;; add data
+        d1 (data/add-data! {:category "long population"
+                            :name "London base population"
+                            :publisher (:id user1)
+                            :s3-url "https://s3.eu-central-1.amazonaws.com/witan-test-data/Long+Pop.csv"
+                            :model-id (:model_id m1)})
```
Comment (acron0) [inline src/witan/app/test_data.clj]: Challenging my own understanding again, this data item seems to refer to `m1` and a category of `"long population"`, although `m1` doesn't specify this category so...is that a mistake or do I misunderstand?

Comment (elisehuard) [inline src/witan/app/test_data.clj]: my mistake, should've been m3 indeed. need to add a check whether the category exists

Comment (elisehuard) [inline src/witan/app/test_data.clj]: actually, referring to the model id in the data is probably a really bad idea.  category should be the key on that one, not the model, and the check on category can happen when the data is added to model or forecast.

Comment (acron0) [inline src/witan/app/test_data.clj]: I wholeheartedly agree.

**Comment by acron0** [comment]
lgtm :shipit:

**Pull request merged**